components/raftstore/src/store/txn_ext.rs

```diff
@@ -244,12 +244,20 @@ impl PeerPessimisticLocks {
         // Locks that are marked deleted still need to be moved to the new regions,
         // and the deleted mark should also be cleared.
         // Refer to the comment in `PeerPessimisticLocks` for details.
-        let removed_locks = self.map.drain_filter(|key, _| {
-            let key = &**key.as_encoded();
+        // There is no drain_filter for BtreeMap, so extra clone are needed.
```

Comment:
`drain_filter` has been renamed to `extract_if` in newer versions. I see BTreeMap has `drain_filter` in the version TiKV uses.

Reply:
`extract_if` is still a nightly-only experimental api(https://github.com/rust-lang/rust/issues/70530) and seems unstable. like `drain_filter` maybe it's difficult to implement such thing on the `BtreeMap`.

Reply:
Considering that:
* TiKV has usages of `Vec` and `HashMap`'s `extract_if` method though they are also exparimental
* Currently `retain` is actually implemented by `extract_if`
Actually I think perhaps introducing an usage of another experimental API might be an acceptable choice 🤔

Reply:
@MyonKeminta 
Sorry I did not find the `#![feature(btree_extract_if)]` related usages in the repo, and the `retaint` is internal implemented by `DrainFilterInner`.
For `HashMap` it is much easier to implement drain filter.

Reply:
I didn't mean that there are any existing use of `btree_extract_if`, but just considering whether it's acceptable to introduce it.

> and the retaint is internal implemented by DrainFilterInner.

Ok.. sorry, guess I was reading a different version